### ISSUE TITLE
Callback for listen method (of NestFactory.create) doesn't get called on v6

### ISSUE THREAD

**Issue by appsolutegeek** [nestjs/nest#1820]
<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
 The callback for the listen never gets called but the application works.

This is for v6.

On v5 it does work.

On v5 is outputs and v6 it does not

## Expected behavior

Callback should be called on the listen method as it did in v5.

## Minimal reproduction of the problem with instructions

This is an example repo that shows the issue. See the console.log inside the callback.

https://github.com/appsolutegeek/error-nest-listener

## Environment

<pre><code>
Nest version: 6
 
For Tooling issues:
- Node version: 11
- Platform: Mac

Others:
</code></pre>

**Comment by ultraMIND** [comment]
In the actual implementation you have to give a second parameter (which is the hostname) to the listen function before the callback, like this: 

`await app.listen(3000, null, () => {
    console.log('listening');
  });`

So it is either a bug or they forgot the remove the deprecated function signature.

**Comment by appsolutegeek** [comment]
Great! Thanks, yes, i passed undefined to the 2nd parameter and now the call is called.

I am going to leave this open as the second overload is there - so it should work. 

The 2nd parameter is important :-) ?

I am passing undefined so i presume it will listen on every host name ?

**Comment by nartc** [comment]
@appsolutegeek If you don't want to pass `null` for the 2nd parameter, you can always resolve back to using `then` on the `Promise` returned by `app.listen()`

```javascript
app.listen(<port>)
    .then(_ => {
         // Do something
    });
```

**Comment by appsolutegeek** [comment]
@nartc Thanks, yes that will work. Cheers. Although I am bit of a Async/Await guy :-) aha ha. Anyway I notice PRs open. Its something I can live with right now.

Cheers!

**Comment by kamilmysliwiec** [comment]
Fixed in 6.0.3. Thanks for reporting!
